### PR TITLE
CASSANDRA-12926 - rejected execution exception logged when task already completed

### PR DESCRIPTION
**Pull request by ayudovin**
this [improvement](https://issues.apache.org/jira/browse/CASSANDRA-12926)

### CODE REVIEW

File: src/java/org/apache/cassandra/repair/SnapshotTask.java [hunk 1]
```diff
@@ -77,7 +77,10 @@ public void response(MessageIn msg)
         public void onFailure(InetAddressAndPort from, RequestFailureReason failureReason)
         {
             //listener.failedSnapshot();
-            task.setException(new RuntimeException("Could not create snapshot at " + from));
+            if (!task.isDone())
+            {
+                task.setException(new RuntimeException("Could not create snapshot at " + from));
+            }
```
Comment (michaelsembwever) [inline src/java/org/apache/cassandra/repair/SnapshotTask.java]: (test for INFRA-16879 .please ignore)

**Comment by michaelsembwever** [comment]
(test for INFRA-16879 .please ignore)

**Comment by ayudovin** [comment]
@michaelsembwever, It's my first pull request to Cassandra and it'll be great if you help me because i'm not sure that i understand  what you mean in this comment "(test for INFRA-16879 .please ignore)"

**Comment by michaelsembwever** [comment]
@ayudovin apologies, these comments are just to test a change in the infrastructure to the cassandra project: https://issues.apache.org/jira/browse/INFRA-16879
once i sort that out i will delete these comments, and then i can help you with the actual PR. again apologises for hijacking your PR.

**Comment by ayudovin** [comment]
@michaelsembwever ok, no problem, thank you

**Comment by Humbedooh** [comment]
(re-test for infra)

File: src/java/org/apache/cassandra/repair/SnapshotTask.java (hunk 1 above)
Comment (michaelsembwever) [reply src/java/org/apache/cassandra/repair/SnapshotTask.java]: (test for INFRA-16879 .please ignore)

**Comment by michaelsembwever** [comment]
(test for INFRA-16879 .please ignore)

**Pull request closed without merging**